docs/architecture/adr-0012-spark-execution-environment.md

```diff
@@ -0,0 +1,42 @@
+# 0012 - Spark Execution Environment
+
+## Context
+
+Part of the Raster Foundry processing workflow requires a source imagery ingest into a GeoTrellis Catalog. GeoTrellis depends on Apache Spark to parallelize work during an ingest, and that work typically occurs on an Apache Spark enabled compute cluster.
+
+Prior iterations of Raster Foundry attempted ingests with an Apache YARN managed compute cluster via Amazon Elastic MapReduce (EMR) on a per-request basis. Unfortunately, that didn't perform well due to the overhead of bootstrapping an EMR cluster.
+
+Our goal for the this iteration aims to keep some Spark cluster components active at all times. This approach helps minimize cluster bootstrapping durations, but also comes at the price of always having some amount of cost-incurring cluster resources on standby.
```

Comment:
>Our goal for the this iteration aims to keep some Spark cluster components active at all times.

Although this is fleshed out more fully below, I think it would be useful to the summary to give a very brief description of the infrastructure you're envisioning (i.e. "Our goal for the this iteration aims to keep some Spark cluster components active at all times by scaling `core` and `task` instances using CloudWatch metrics") .